### ISSUE TITLE
Regression Bug in `torch.nn.functional.interpolate`

### ISSUE THREAD

**Issue by TobyRoseman** [pytorch/pytorch#71188]
### 🐛 Describe the bug

```python
import torch

torch.manual_seed(0)
x = torch.rand((1, 3, 10, 10))
y = torch.nn.functional.interpolate(x, (1,1), align_corners=False, mode='bilinear')
print(y)
```
------

PyTorch `1.9.1` outputs:
```
tensor([[[[0.4963]],

         [[0.9704]],

         [[0.5210]]]])
```
I believe this is the correct result. [The Core ML Framework](https://developer.apple.com/documentation/coreml) produces the same result given the same input.

------

PyTorch `1.10.1` outputs:
```
tensor([[[[0.3554]],

         [[0.2755]],

         [[0.3230]]]])
```
This is seems wrong.

### Versions

Collecting environment information...
PyTorch version: 1.10.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.1 (x86_64)
GCC version: Could not collect
Clang version: 13.0.0 (clang-1300.0.29.3)
CMake version: version 3.12.1
Libc version: N/A

Python version: 3.7.11 (default, Jul 27 2021, 07:03:16)  [Clang 10.0.0 ] (64-bit runtime)
Python platform: Darwin-21.2.0-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.10.1
[pip3] torchvision==0.11.2
[conda] numpy                     1.19.5                   pypi_0    pypi
[conda] torch                     1.10.1                   pypi_0    pypi
[conda] torchvision               0.11.2                   pypi_0    pypi

cc @ezyang @gchanan @zou3519 @bdhirsh @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by mruberry** [comment]
Thank you for reporting this issue, @TobyRoseman. I believe we would accept a PR fixing this.

**Comment by jbschlosser** [comment]
Looks to be expected behavior from a fix in #61166. Previously, the value of `align_corners` did not matter for this scenario, and now it does. This was apparently done to match the behavior of OpenCV / PIL.

**Comment by jbschlosser** [comment]
Closing as expected behavior. Feel free to reopen if you disagree.
